driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -722,12 +759,30 @@ public boolean onTimeout(Connection connection, long latency, int retryCount) {
             }
 
             Host queriedHost = current;
-            OperationTimedOutException timeoutException = new OperationTimedOutException(connection.address);
+
+            // This cannot be OperationTimedOutException because
```

Comment:
Thanks for the explanation.  So in general, should OperationTimedOutException should only be raised internally for things like preparing statements, checking cluster name, etc.?  I wonder if we should make that clear in the documentation, as it is currently somewhat unclear with regards to that.

Also since we have specific exceptions that inherit DriverException, would it make sense to create a specific implemetation for this case so users can properly check for it?